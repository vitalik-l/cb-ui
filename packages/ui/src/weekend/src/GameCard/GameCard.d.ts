/// <reference types="react" />
declare type Props = {
    className?: string;
    suit?: 'hearts' | 'diamonds' | 'clubs' | 'spades';
    value?: string | number;
};
export declare const GameCard: (props: Props) => JSX.Element;
export {};
