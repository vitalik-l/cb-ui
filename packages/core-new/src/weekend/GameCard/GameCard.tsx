import React from 'react';
import clsx from 'clsx';

// local files
import styles from './WkdGameCard.module.scss';

type Props = {
  className?: string;
  suit?: 'hearts' | 'diamonds' | 'clubs' | 'spades';
  value?: string | number;
};

export const GameCard = (props: Props) => {
  const { className, suit, value, ...restProps } = props;

  return (
    <div className={clsx(styles.root, className)} {...restProps}>
      <div className={styles.back} />
      <div className={clsx(styles.face, !!suit && styles[`suit_${suit}`])}>
        <span className={styles.value}>{value}</span>
        <span className={styles.valueCenter} />
        <span className={styles.value}>{value}</span>
      </div>
    </div>
  );
};
