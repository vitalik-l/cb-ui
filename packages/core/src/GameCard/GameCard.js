import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function GameCard(props) {
  const { suit, value, className } = props;

  return (
    <div className={classNames('cb-GameCard', className)}>
      <span className="back" />
      <span className={`face ${suit}`}>
        <dt>{value}</dt>
        <b />
        <dt>{value}</dt>
      </span>
    </div>
  );
}

GameCard.propTypes = {
  suit: PropTypes.oneOf(['hearts', 'diamonds', 'clubs', 'spades']),
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  className: PropTypes.string,
};

export default GameCard;
