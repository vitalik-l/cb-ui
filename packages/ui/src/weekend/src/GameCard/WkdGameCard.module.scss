$root: WkdGameCard;
$width: 7.5em;
$fontSize: 1em;

@mixin suit($suit) {
  .suit_#{$suit} {
    @if $suit == 'hearts' or $suit == 'diamonds' {
      color: #f00;
    } @else {
      color: black;
    }

    .value:after {
      background: url('./img/#{$suit}.svg');
    }

    .valueCenter:after {
      content: url('./img/#{$suit}.svg');
    }
  }
}

@font-face {
  font-family: Rockwell;
  src: url('fonts/rockwell-bold.woff2') format('woff2'),
    url('fonts/rockwell-bold.woff') format('woff');
  font-weight: 700;
  font-style: normal;
}

:global(.#{$root}) {
  display: inline-block;
  width: $width;
  height: $width * 1.35;
  flex: 0 0 $width + 0.1;
  border: none;
  border-radius: 0.3em;
  box-shadow: 0 0 0.3em 0.1em #666;
  transform: rotateY(179.9deg);
  visibility: visible;
  transform-style: preserve-3d;
  margin: 0;
  font-size: $fontSize;
}

.back {
  background: red url(./img/back.png) no-repeat;
  background-size: cover;
  border: 0.3em solid white;
  border-radius: inherit;
  margin: 0;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  backface-visibility: hidden;
  box-sizing: border-box;
}

.face {
  background-color: #fff;
  transform: rotateY(179.9deg);
  display: flex;
  align-items: center;
  justify-content: center;
  font-family: Rockwell;
  font-size: 1.5em;
  font-weight: bold;
  margin: 0;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  backface-visibility: hidden;
  border: 0.3em solid white;
  border-radius: inherit;
  box-sizing: border-box;
}

.value {
  line-height: 1;
  text-align: center;
  font-size: 1.2em;
  display: inline-flex;
  flex-direction: column;
  height: auto;
  align-self: flex-start;
  justify-content: center;
  align-items: center;

  &:last-of-type {
    transform: rotate(179.9deg);
    align-self: flex-end;
  }

  &:after {
    content: '';
    display: block;
    text-align: center;
    font-size: 1em;
    margin-top: 2px;
    width: 0.8em;
    height: 0.8em;
    background-size: contain;
  }
}

.valueCenter {
  flex: 1;
  text-align: center;

  &:after {
    display: block;
    width: 1.5em;
    height: 1.5em;
    margin: auto;
  }
}

@include suit('spades');
@include suit('clubs');
@include suit('diamonds');
@include suit('hearts');

:export {
  root: $root;
}
