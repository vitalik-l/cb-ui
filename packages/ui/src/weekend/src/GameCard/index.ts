export { GameCard } from './GameCard';
export { GameCard as default } from './GameCard';
