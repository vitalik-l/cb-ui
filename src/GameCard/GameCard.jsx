import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class GameCard extends PureComponent {
  render() {
    const { suit, value, className } = this.props;
    return (
      <div className={className || 'cb-GameCard'}>
        <span className="back" />
        <span className={`face ${suit}`}>
          <dt>{value}</dt>
          <b />
          <dt>{value}</dt>
        </span>
      </div>
    );
  }
}

GameCard.propTypes = {
  suit: PropTypes.oneOf(['hearts', 'diamonds', 'clubs', 'spades']),
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

export default GameCard;
