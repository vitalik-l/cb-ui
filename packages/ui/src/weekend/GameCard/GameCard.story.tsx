import React from 'react';
import { Story } from '@storybook/react';

// local files
import { createStory } from '../createStory';
import { GameCard } from './GameCard';

const story = createStory({
  title: 'GameCard',
  component: GameCard,
  argTypes: {
    value: { control: 'text' },
    back: { control: 'boolean' },
  },
  args: {
    value: 'A',
    suit: 'clubs',
  },
});

const Template = ({ back, ...args }: any) => {
  return <GameCard {...args} style={back ? { transform: 'rotateY(0)' } : undefined} />;
};

export const Default: Story = Template.bind({});
Default.storyName = 'GameCard';

export default story;
