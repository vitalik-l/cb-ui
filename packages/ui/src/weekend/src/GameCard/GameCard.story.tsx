import React from 'react';
import { Story } from '@storybook/react';

// local files
import { index } from '../../story/createStory';
import { GameCard } from './GameCard';

const story = index({
  title: 'GameCard',
  component: GameCard,
  argTypes: {
    value: { control: 'text' },
    back: { control: 'boolean' },
  },
  args: {
    value: 'A',
    suit: 'clubs',
  },
});

const Template = ({ back, ...args }: any) => {
  return <GameCard {...args} style={back ? { transform: 'rotateY(0)' } : undefined} />;
};

export const Default: Story = Template.bind({});
Default.storyName = 'GameCard';

export default story;
