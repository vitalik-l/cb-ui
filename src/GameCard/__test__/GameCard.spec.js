import React from 'react';
import {shallow} from 'enzyme';
import GameCard from '../GameCard';

describe('GameCard', () => {
    const wrapper = shallow(<GameCard/>);
    it('custom className. Should set custom className prop to root element', () => {
        wrapper.setProps({className: 'test'});
        expect(wrapper).toMatchSnapshot();
    });
});