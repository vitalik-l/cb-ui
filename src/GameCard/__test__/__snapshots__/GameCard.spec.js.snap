// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCard custom className. Should set custom className prop to root element 1`] = `
<div
  className="test"
>
  <span
    className="back"
  />
  <span
    className="face undefined"
  >
    <dt />
    <b />
    <dt />
  </span>
</div>
`;
